WPFHospitalEditor/HospitalMap.xaml.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
+using WPFHospitalEditor.MapObjectModel;
+
+namespace WPFHospitalEditor
+{
+    /// <summary>
+    /// Interaction logic for HospitalMap.xaml
+    /// </summary>
+    public partial class HospitalMap : Window
+    {
+        AllMapObjects allMapObjects = new AllMapObjects();
+        public HospitalMap()
+        {
+            InitializeComponent();
+            addObjectToCanvas(AllMapObjects.allOuterMapObjects);
+        }
+        private void selectBuilding(object sender, MouseButtonEventArgs e)
+        {
+            MapObject chosenBuilding = checkWhichObjectIsClicked(e, AllMapObjects.allOuterMapObjects);
+            if (chosenBuilding != null)
+            {
+                goToClickedBuilding(chosenBuilding);
+            }
+        }
+        public MapObject checkWhichObjectIsClicked(MouseButtonEventArgs e, List<MapObject> allMapObjectsShowed)
+        {
+            for (int i = 0; i < allMapObjectsShowed.Count; i++)
+            {
+                if (checkIfPointIsInRectangle(e, allMapObjectsShowed[i]))
+                {
+                    return allMapObjectsShowed[i];
+                }
+            }
+            return null;
```

Comment:
Whether an object can be assigned a click event instead of manually determining which object is clicked